cpp/tests/test_chunk.cpp

```diff
@@ -40,15 +40,16 @@ namespace {
 PackedData create_packed_data(
     cuda::std::span<uint8_t const> metadata,
     cuda::std::span<uint8_t const> data,
-    rmm::cuda_stream_view stream
+    rmm::cuda_stream_view stream,
+    BufferResource* br
 ) {
     auto metadata_ptr =
         std::make_unique<std::vector<uint8_t>>(metadata.begin(), metadata.end());
     auto data_ptr = std::make_unique<rmm::device_buffer>(data.size(), stream);
     RAPIDSMPF_CUDA_TRY(
         cudaMemcpy(data_ptr->data(), data.data(), data.size(), cudaMemcpyHostToDevice)
     );
-    return PackedData{std::move(metadata_ptr), std::move(data_ptr)};
+    return PackedData{std::move(metadata_ptr), std::move(data_ptr), stream, br};
```

Comment:
I _think_ I would prefer if the `PackedData` ctor were trivial and the conversion into `Buffer` (from `device_buffer`)  was explicit. i.e. here:

```
auto buf = br->move_from_device_buffer(data_ptr, stream);r

return PackedData{std::move(metadata), std::move(buf)};
```
WDYT?

Reply:
good idea, done